inject/src/main/java/io/micronaut/inject/annotation/RequiresValidation.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.inject.annotation;
+
+import io.micronaut.core.annotation.Internal;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.SOURCE;
+
+/**
+ * Internal method marks a type, method or a field for validation.
+ *
+ * @author Denis Stepanov
+ * @since 4.0.0
+ */
+@Documented
+@Retention(SOURCE)
+@Target({ElementType.METHOD, ElementType.TYPE, ElementType.FIELD})
+@Internal
+public @interface RequiresValidation {
```

Comment:
Move this annotation to another package so if you `*` import `inject.annotation` which includes a lot of public API it doesn't appear